docs/rule/no-whitespace-within-word.md

```diff
@@ -46,6 +46,10 @@ This rule **allows** the following:
 
 This rule uses the heuristic of letter, whitespace character, letter, whitespace character, letter which makes it a good candidate for most use cases, but not ideal for some languages (such as Japanese).
 
+## Configuration
+
+* object - { 'character-regex': '[a-zA-Z]' } -Specify a custom regex (as string) to identify single characters. The default matches uppercase & lowercase characters.
```

Comment:
Could it be overkill to have an option for this? I feel like this rule should just work and shouldn't require customization to handle basic use cases.

Reply:
My idea was to account to be able to opt-in for non-latin characters, e.g. Chinese or Arabic (although I guess for those more symbol-based locales it makes sense to _not_ check them anyhow, but not an expert on that by any means) or something like German Umlaute, if somebody really needs this. But I'm also happy to remove the config option!

Reply:
I think it's a nice goal for this rule to be able to handle different languages, but I don't think this option is practical currently. What are the chances someone using a different language is going to formulate the regexp for their own language to provide to this option? Pretty low I suspect.

The rule doc currently mentions `but not ideal for some languages` so I'd rather just focus on English now and omit this obscure option which.

That said, if there are more user-friendly ways we can make this rule generic and able to handle different languages, I'm all for it.

Reply:
:+1: OK, I removed the config option :) IMHO it should be good enough for most cases, and it should, if, only produce false negatives, not false positives, which is OK for a linting rule like this I'd say anyhow.

Reply:
We can consider an option or other improvements later if needed.